Kernel/Interrupts/InterruptManagement.cpp

```diff
@@ -180,7 +180,10 @@ UNMAP_AFTER_INIT void InterruptManagement::switch_to_ioapic_mode()
         m_pci_interrupt_overrides = mp_parser->get_pci_interrupt_redirections();
     }
 
-    APIC::the().init_bsp();
+    if (!APIC::initialized()) {
```

Comment:
Is there a reason to believe someone will initialize the APIC before this call? I would say that might be a bug we want to catch, so maybe `VERIFY` statement is better placed here.

Reply:
I was trying to be defensive. But `APIC::initialize` already has the `VERIFY`. So, I will just remove the if condition.